src/service/resources-impl.js

```diff
@@ -248,6 +248,14 @@ export class Resources {
   }
 
   /**
+   * Signals that the document has been started rendering.
+   * @restricted
+   */
+  renderStarted() {
```

Comment:
I'm working on moving some scheduling here as well.

This should be called and work exactly the same with amp-inabox with one exception: we also want to postMessage `render-start` which is what you are working on :) Where exactly postMessage belongs in, I'm not sure. I was going to see where you end up putting it. But my first suggestion was to stick it right into `amp-inabox.js`. That means that we may need a callback from this method, which you can add later if that's what you'd like to do.